### PR TITLE
Jenish/feature/finish-iris-obc-interfacing

### PR DESCRIPTION
**Pull request by JenishPatel99**
Completed iris-obc interface required for flight functionality. There is a bit of refactoring to be done, but it was necessary to update the main branch to include successfully working Iris

### CODE REVIEW

File: Core/Inc/iris_system.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ extern "C" {
 
 //#define UART_DEBUG
 #define SPI_DEBUG
+#define SPI_DEBUG_UART_OUTPUT
```
Comment (LiamDroog) [inline Core/Inc/iris_system.h:52]: image

File: Core/Src/housekeeping.c [hunk 2]
```diff
@@ -19,6 +19,7 @@ housekeeping_packet_t _get_housekeeping() {
     hk.gate_temp = get_temp(GATE_TEMP_SENSOR);
     hk.imagenum = get_image_num(1);
     hk.software_version = software_ver;
+    hk.errornum = 0x00;
```
Comment (LiamDroog) [inline Core/Src/housekeeping.c:22]: Can't have any errors if the error number is hardcoded to 0 amiright

File: Core/Src/main.c [hunk 3]
```diff
@@ -423,7 +423,7 @@ static void MX_SPI2_Init(void) {
     hspi2.Init.Direction = SPI_DIRECTION_2LINES;
     hspi2.Init.DataSize = SPI_DATASIZE_8BIT;
     hspi2.Init.CLKPolarity = SPI_POLARITY_LOW;
-    hspi2.Init.CLKPhase = SPI_PHASE_1EDGE;
+    hspi2.Init.CLKPhase = SPI_PHASE_2EDGE;
```
Comment (LiamDroog) [inline Core/Src/main.c:426]: Why are you switching the NAND flash SPI? It should be rising edge iirc

File: Core/Src/spi_command_handler.c [hunk 4]
```diff
@@ -3,13 +3,17 @@
...
+#include <spi_bitbang.h>
 
 extern SPI_HandleTypeDef hspi1;
 extern uint8_t cam_to_nand_transfer_flag;
 
-uint32_t image_length = 0x5F3; // Only here for testing purposes
+uint32_t image_length; // Only here for testing purposes
 static uint32_t count = 0x0FFF0000;
 
+#define VIS 0x3C
```
Comment (LiamDroog) [inline Core/Src/spi_command_handler.c:14]: better to use VIS_SENSOR and NIR_SENSOR as defined in `iris_system.h`

File: Core/Src/spi_command_handler.c [hunk 5]
```diff
@@ -130,12 +134,23 @@ int spi_handle_command(uint8_t obc_cmd) {
         uint8_t buffer[sizeof(hk)];
         memcpy(buffer, &hk, sizeof(hk));
         spi_transmit(buffer, sizeof(buffer));
-
         return 0;
     }
     case IRIS_TAKE_PIC: {
         // needs dedicated thought put towards implement
-        arducam_capture_image(0);
+        arducam_set_resolution(JPEG, 640, VIS);
```
Comment (LiamDroog) [inline Core/Src/spi_command_handler.c:141]: Not a great place for setting the resolution. That should be done on boot. If we want timing accuracy we shouldn't be writing a flurry of I2C regs when we should be taking a picture

**Review by LiamDroog — CHANGES_REQUESTED** [review]

**Comment by LiamDroog** [comment]
Also the checks fail, so we should look into that

File: Core/Src/housekeeping.c (hunk 2 above)
Comment (JenishPatel99) [reply Core/Src/housekeeping.c:22]: Yes, although I was not sure where was the error counter

File: Core/Src/main.c (hunk 3 above)
Comment (JenishPatel99) [reply Core/Src/main.c:426]: Oops, my bad

File: Core/Src/spi_command_handler.c (hunk 5 above)
Comment (JenishPatel99) [reply Core/Src/spi_command_handler.c:141]: Agreed, I will make sure that all sensor reg information is pre-defined during boot

File: Core/Inc/iris_system.h (hunk 1 above)
Comment (JenishPatel99) [reply Core/Inc/iris_system.h:52]: This was meant to be there for debugging only, I need to see what iris is doing when interacting with obc

File: Core/Src/housekeeping.c (hunk 2 above)
Comment (LiamDroog) [reply Core/Src/housekeeping.c:22]: I haven't implemented it yet. I'll implement it alongside our audit log

**Review by Taylorr82 — APPROVED** [review]

File: Core/Src/housekeeping.c (hunk 2 above)
Comment (JenishPatel99) [reply Core/Src/housekeeping.c:22]: sounds good

**Review by LiamDroog — APPROVED** [review]

**Pull request merged**